Automattic/jetpack
# Podcast Player: Clean-up styles & fix few remaining issues

Built on top of #15504 as a part of completing the remaining Podcast Player improvements. My main goal was to go through styles.scss and clean it up a bit to minimize technical debt. On the way, I found a few small bugs and decided not to create separate issues for them as they kind of fixed themselves while cleaning up the code :)

#### Changes proposed in this Pull Request:

Maintenance:

- Removed duplicated SCSS variables,
- Renamed some SCSS variables for clarity,
- Used available SCSS variables for hardcoded values where possible,
- Unified gutter and some paddings into simple `$gutter-s/m/l` vars,
- Cleaned up and made colors styles more consistent and easy to follow,
- Cleaned up a few misplaced comments.

Bugfixes:

- Made links styling consistent (Podcast title, Track title, Error message)
   Links have the Secondary color initially and Primary when hovered (except for an active Track, which has Primary color at all times). Previously that was true only when default colors were applied. Now it's true also for any custom colors thanks to progressive enhancement with the CSS vars. Note that therefore hover colors are not supported in, i.e. IE11.
- Fixed error message link ("Open in a new tab") to which custom colors weren't applied in browsers that don't support CSS vars (IE11).
- Used correct default, `$dark-gray-300` (Secondary) color for episode description text instead of `$dark-gray-500`.
- Made the ME.js progress bar tooltip's text color the same as the background for proper contrast.

One _enhancement_:

- Removed block border (p1587552447434800-slack-ajax)

#### Testing instructions:

- Add the Podcast Player block (use e.g. this URL https://anchor.fm/s/9400d7c/podcast/rss),
- Make sure that podcast title, inactive track and the "Open in a new tab" track error links have all correct colors - Secondary initially and Primary on hover (hover colors applied only for browsers that support CSS variables),
- Make sure that episode description's default color is correct (`#6c7781`),
- Make sure the block has no border,
- Make sure the ME.js progress bar tooltip's text color is the same as the background,
- All of the above should be true when switching themes and global styles.

Demo GIF:
- Colors: custom
- Theme: Varia
- Global styles: "System Font & Libre Baskerville"

![Apr-22-2020 16-00-44](https://user-images.githubusercontent.com/1451471/79991786-fbb1cc80-84b2-11ea-8aa6-75feac1d15c8.gif)

IE11 screenshots (same for editor and front-end):

| playback | track error |
| ------------- | ------------- |
|<img width="598" alt="Screenshot 2020-04-23 13 14 00" src="https://user-images.githubusercontent.com/1451471/80093462-d2ea0f80-8564-11ea-99e0-24ebc6f59923.png">| <img width="598" alt="Screenshot 2020-04-23 13 15 34" src="https://user-images.githubusercontent.com/1451471/80093470-d67d9680-8564-11ea-9590-27a9588bfd34.png">|


No-JS screenshot (front-end):
- Colors: custom
- Theme: Varia

<img width="604" alt="Screenshot 2020-04-23 12 53 25" src="https://user-images.githubusercontent.com/1451471/80091610-b39db300-8561-11ea-93b6-99afd8285011.png">



#### Proposed changelog entry for your changes:
* none
